components/collector/src/source_collectors/azure_devops/lead_time_for_changes.py

```diff
@@ -0,0 +1,62 @@
+"""Azure DevOps Server lead time for changes collector."""
+
+from statistics import mean
+from typing import cast
+
+from dateutil.parser import parse
+
+from collector_utilities.functions import days_ago
+from collector_utilities.type import Value
+from model import Entity, Entities, SourceResponses
+
+from .issues import AzureDevopsIssues
+
+
+class AzureDevopsLeadTimeForChanges(AzureDevopsIssues):
+    """Collector to calculate lead time for changes from Azure Devops Server."""
+
+    def _item_select_fields(self) -> list[str]:
+        """Also request date fields to calculate lead time."""
+        base_fields = super()._item_select_fields()
+        return base_fields + ["System.CreatedDate", "System.ChangedDate"]
+
+    def _include_issue(self, issue: dict) -> bool:
+        """Return whether this issue should be counted."""
+        if not issue["fields"]["System.State"] in ["Closed", "Done"]:
+            return False
+
+        finished_date = issue["fields"].get("System.ChangedDate")  # TODO - how to find out System.State update ?
+        if not finished_date:
+            return False
+
+        return days_ago(parse(finished_date)) <= int(cast(str, self._parameter("lookback_days")))
+
+    async def _parse_value(self, responses: SourceResponses) -> Value:
+        work_items = (await self._work_items(responses))
+        if not work_items:
+            return None
+
+        lead_times = [self.__lead_time(item) for item in work_items]
+        return str(round(mean(lead_times)))
```

Comment:
I would do this. Maybe we need to decide how many lines of indented code warrants a guard clause ;-)

```suggestion
        if work_items := (await self._work_items(responses)):
            lead_times = [self.__lead_time(item) for item in work_items]
            return str(round(mean(lead_times)))
        return None
```

Reply:
Old habits for big data analysis code die hard :))